### DISCUSSION TITLE
Neo-tree and ahmedkhalf/project.nvim how to make them work together?

### DISCUSSION THREAD

**Discussion by CyrilPeponnet** [nvim-neo-tree/neo-tree.nvim#728] [Q&A]
Hey guys,

I moved from nvim-tree to neo-tree and looks like the handle of cwd / tcd / lcd is not as smooth.

The "ahmedkhalf/project.nvim" plugin change the directory when loading a project and has the following options:

```
  -- What scope to change the directory, valid options are
  -- * global (default)
  -- * tab
  -- * win
  scope_chdir = 'global',
```

On the other side neo-tree have for filesystem source:

```
       	bind_to_cwd = true,
        cwd_target = {
          sidebar = "tab",   -- sidebar is when position = left or right
          current = "window" -- current is when position = current
        },
```

(and tbh I cannot understand the cwd target ones ;p)

So now here is the thing:

Given I have a file foo in project A (with its own root) and a file bar in a project B (different root).

If I use the bind_to_cwd to true, loading the file foo and open the tree will open the tree in the project A root dir. Now If I open the file bar from project B (invoking the project plugin) with the tree still open, the tree is properly "revealed" to tree B for a split second then goes back to the root A. If  toggle the tree off and on again with force reveal flag it goes back to B but whenever I click on my buffer that contain the bar file from B the tree "refresh" to root A.

So far to have a behavior that pseudo work I need to disable the bind_to_cwd but in that case I need to manually refresh reveal the tree.

Is there a way to make it work properly with the settings above?

For now I have a hack like:

```lua
-- -- Refresh the neo-tree if the cwd changed
vim.api.nvim_create_autocmd("BufEnter", {
	pattern = { "*" },
	callback = function()
		local path = require("util").get_root()
		local tstate = require("neo-tree.sources.manager").get_state("filesystem")
		if not tstate.path then
			return
		end
		-- check if the tree is open
		if require("neo-tree.ui.renderer").window_exists(tstate) and path ~= tstate.path then
			require("neo-tree.sources.manager").navigate("filesystem", path)
		end
	end,
})
```

Thanks :)

**Comment by cseickel** [comment]
I don't know if it will work or not, but based on the choices presented the correct choice for `project.nvim` would be:
```
 scope_chdir = 'tab',
```

That will line up with how neo-tree manages the working directory.

**Reply by CyrilPeponnet** [reply]
I tried that with no luck. I have the same split second tree refresh that goes back to project A.

I can live with my workaround though was just wondering if I missed something.
